### PR TITLE
complete template for article and moderncv class | safer bash settings

### PR DESCRIPTION
**Pull request by toogley**
Sorry for Commit, `ff5b2e1` I forgot to pull your changes in. Unfortunately, i currently don't know how to "reorder" that commit so that it doesn't appear in this PR, without rewriting all other Commits. Does this issue bother you?

### LaTeX

I don't know if it would be a valuable addition to provide full templates (specifically, I don't know whether Emacs has better options for automatically inserting templates when a new file is created). What do you think?

### Bash

source: http://redsymbol.net/articles/unofficial-bash-strict-mode/
Short Explanation:

    set -e               # fail if any command has non-zero exit status
    set -u               # fail if unbound variables are used
    set -o pipefail      # fail if command in a pipe isn't successful
    IFS=$'\n\t'          # wordsplitting will happen only on \n and \t, not on spaces
                         # => would otherwise break "My Song.mp3" or "Benjamin Franklin"

### CODE REVIEW

**Comment by npostavs** [comment]
> Unfortunately, i currently don't know how to "reorder" that commit so that it doesn't appear in this PR, without rewriting all other Commits.

You could rebase, then `push -f` the result your branch.

**Comment by toogley** [comment]
@npostavs  to cite from the git rebase manpage: 

> The todo list presented by --preserve-merges --interactive does not represent the topology of the revision graph. Editing commits and rewording their commit messages should work fine, but **attempts to reorder commits tend to produce counterintuitive results.**

**Comment by npostavs** [comment]
In this case you don't want to preserve the merge anyway.

**Comment by toogley** [comment]
Hm, that is at least a little improvement. Maybe i can improve that even further later.  @npostavs thanks for the tip.

**Comment by npostavs** [comment]
You seems to have duplicated some other people's commits. Try an interactive rebase and mark those commits with `#` to get rid of them.

**Comment by toogley** [comment]
Thanks, i didn't knew that feature.

File: sh-mode/safe-bang [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# -*- mode: snippet -*-
+# name: safer bash settings for scripts
+# key: s!
+# --
+#!/usr/bin/env bash
+set -euo pipefail
```
Comment (AndreaCrotti) [inline sh-mode/safe-bang]: this script is maybe not that generic but in any case it definitively doesn't harm and can be good to have..

**Comment by AndreaCrotti** [comment]
Ok great thanks, in the future please do two different Pull requests (I mean splitting the bash changes from the latex changes)

File: sh-mode/safe-bang (hunk 1 above)
Comment (toogley) [inline sh-mode/safe-bang]: Can you please explain what you meant by "generic"? I think every bash script does profit from such settings.

**Pull request merged**